File: client_angular/src/app/models/avis.ts
export interface Avis {
  id: number;
  commentaire: string;
  note: number;                 // entre 1 et 5
  dateCreation: string;        // format ISO (ex: '2025-08-01T14:30:00')
  utilisateurId: number;       // ID de l'utilisateur
  utilisateurNom: string;       // ID de l'utilisateur
  vehiculeId: number;          // ID du véhicule
  vehiculeMarque: string;
  vehiculeModele: string;
}

File: springboot/app/src/main/java/com/bugshan/automative/group/app/model/Avis.java
package com.bugshan.automative.group.app.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "avis")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Avis {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(length = 500)
    private String commentaire;

    @Column(nullable = false)
    private int note;  // Entre 1 et 5

    @Column(name = "date_creation", nullable = false)
    private LocalDateTime dateCreation;

    @ManyToOne(fetch = FetchType.EAGER)
    @JsonBackReference("user")
    @JoinColumn(name = "utilisateur_id")
    private Utilisateur utilisateur;

    @ManyToOne(fetch = FetchType.EAGER)
    @JsonBackReference("vehicle")
    @JoinColumn(name = "vehicule_id")
    private Vehicule vehicule;
}
